lib/gcloud/resource_manager/service.rb

```diff
@@ -0,0 +1,108 @@
+# Copyright 2016 Google Inc. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+require "gcloud/version"
+require "google/apis/cloudresourcemanager_v1beta1"
+
+module Gcloud
+  module ResourceManager
+    ##
+    # @private
+    # Represents the service to Resource Manager, as well as expose the API
+    # calls.
+    class Service
+      ##
+      # Alias to the Google Client API module
+      API = Google::Apis::CloudresourcemanagerV1beta1
+
+      attr_accessor :credentials
+
+      ##
+      # Creates a new Service instance.
+      def initialize credentials
+        @credentials = credentials
+        @service = API::CloudResourceManagerService.new
+        @service.client_options.application_name    = "gcloud-ruby"
+        @service.client_options.application_version = Gcloud::VERSION
+        @service.authorization = @credentials.client
+      end
+
+      def service
+        return mocked_service if mocked_service
+        @service
+      end
+      attr_accessor :mocked_service
+
+      ##
+      # Returns API::ListProjectsResponse
+      def list_project filter: nil, token: nil, max: nil
```

Comment:
It could be. I wasn't trying to make improvements in that commit, just convert the client.